Automattic/pocket-casts-android
# Update watch log in screens to include log in on phone

## Description
This is the first PR toward updating the log in screens for the watch app. This PR replaces the single (placeholder) email login option with a screen that lists google sign-in, log in on phone, and log in with email options. 

Log in with Google is not implemented. This will be implemented in a separate PR.

Log in with phone includes a screen with directions for the user.

Log in with email is still using placeholder screens. That will be properly implemented in a separate PR.

## Testing Instructions
1. Open the app without being logged in
2. Open settings
3. Tap on the Log in button
4. Tap on Log in on phone
5. Observe a screen with instructions for logging in on the phone
6. Tap the screen to go back
7. Log into the app (either by using the placeholder login with email button or by logging in on a paired phone)
8. Observe that the settings screen now has a log out button.

## Screenshots or Screencast 

https://user-images.githubusercontent.com/4656348/235274466-2c2d9f8d-f869-4442-be0d-9d3088d3e6d9.mov

## Checklist
